pkg/cache/v3/delta.go

```diff
@@ -22,15 +22,35 @@ import (
 	"github.com/envoyproxy/go-control-plane/pkg/server/stream/v3"
 )
 
+type resourceContainer interface {
+	GetResources(typeURL string) map[string]types.Resource
+
+	GetVersionMap(typeURL string) map[string]string
+
+	GetVersion(typeURL string) string
+}
+
+// This struct exist only because we want to reuse createDeltaResponse function from linear cache code
+// so we create a wrapper, which adopts LinearCche to satisfy ResourceContainer interface
+type cacheWrapper struct {
```

Comment:
What can we do to get rid of this?

Reply:
I think we have 2 options:
1. Make Linear cache to implement the following methods
```
	GetResources(typeURL string) map[string]types.Resource
	GetVersionMap(typeURL string) map[string]string
	GetVersion(typeURL string) string
```
The only problem with this is that typeURL parameter is redundant for linear cache.
2. Change `createDeltaResponse` signature to something like this.
```
func createDeltaResponse(ctx context.Context, req *DeltaRequest, state stream.StreamState, resources map[string]types.Resource, versionMap map[string]string, systemVersion string, log log.Logger) *RawDeltaResponse
```
then in the `respondDeltaLinear ` and `respondDeltaSnapshot` function get the needed parameters either from snapshot or from the cache. 

I think both options are ok.

Reply:
We only use it [here](https://github.com/s-matyukevich/go-control-plane/blob/incremental-linear/pkg/cache/v3/delta.go#L134) and yes, it is used only for debugging purposes (we only need it to send it with the DeltaDiscoveryResponse to envoy)

Reply:
> We only use it here and yes, it is used only for debugging purposes (we only need it to send it with the DeltaDiscoveryResponse to envoy)

Okay great I just wanted to make sure we didn't do anything special in linear

Reply:
I'm sort of inclined to have the Linear Cache implement the three methods above so we can remove the cache wrapper, and then just ignore the type url params

Reply:
> You mean they just won't be used right since the linear cache is instantiated on a per resource type basis

Yes, exactly

> I'm sort of inclined to have the Linear Cache implement the three methods above so we can remove the cache wrapper, and then just ignore the type url params. Do you think that would reduce the boilerplate?

ok, I can do that.

Reply:
Actually this caused problems: linear cache already has method `GetResources`, which doesn't accept typeUrl parameter. Of cause, we can add another method `GetResourcesDelte(typeUrl string)` but I don't think this is acceptable for a public facing API. 

I actually implemented a little bit different solution, which still allows us to get rid of the cacheWrapper. Let me know what do you think.

Reply:
I like it! I think we can roll with what we've got